pkgs/development/tools/database/sqldeveloper/default.nix

```diff
@@ -0,0 +1,48 @@
+{ stdenv, makeWrapper, requireFile, unzip, oraclejdk7}:
+
+stdenv.mkDerivation rec {
+  version = "4.0.3.16.84";
+  name = "sqldeveloper-${version}";
+
+  src = requireFile {
+    name = "${name}-no-jre.zip";
+    url = http://www.oracle.com/technetwork/developer-tools/sql-developer/downloads/index.html;
+    sha256 = "1qbqjkfda7xry716da2hdbbazks96rgyslrw1lw0azmqdp1mir7g";
+  };
+
+  buildInputs = [ makeWrapper unzip ];
+
+  buildCommand = ''
+    mkdir -p $out/bin
+    # patch to be able to install a sqldeveloper wrapper script compliant with nix's bin folder once installed
+    echo -e '#!/bin/bash\ncd "`dirname $0`"/../sqldeveloper/bin && bash sqldeveloper $*' >> $out/bin/sqldeveloper
+
+    cd $out
+    unzip ${src}
+    cp -r sqldeveloper/* $out/
+    # Activate the needed shell script
+    rm $out/sqldeveloper.sh
+    chmod +x $out/bin/sqldeveloper
+    chmod +x $out/sqldeveloper/bin/sqldeveloper
+
+    wrapProgram $out/bin/sqldeveloper \
+      --set JAVA_HOME "${oraclejdk7}"
```

Comment:
Also, here is the message when installed without the JAVA_HOME set is this:

``` sh
# tony at lenovo in ~/repo/perso/nixpkgs on git:add-sqldeveloper-derivation x [12:20:08]
$ ./result/bin/sqldeveloper

 Oracle SQL Developer
 Copyright (c) 1997, 2014, Oracle and/or its affiliates. All rights reserved.

Type the full pathname of a JDK installation (or Ctrl-C to quit), the path will be stored in /home/tony/.sqldeveloper/4.0.0/product.conf
```

So it asks for a jdk.
But after modification, the jre seems to work fine (at least for the startup).